packages/daemon/src/generator/boa.ts

```diff
@@ -0,0 +1,35 @@
+import { GenerateOptions } from '../service/pipeline';
+import * as fs from 'fs-extra';
+import { JobEntity } from '../model/job';
+import * as path from 'path';
+
+export function boaGenerator(job: JobEntity, projPackage: any, dist: string, opts: GenerateOptions) {
```

Comment:
Shall we have a common interface for the generator function?

Reply:
Different generators might have their own parameter. I.E. tvm requires model type while boa does not